hugegraph-core/src/main/java/org/apache/hugegraph/util/StringEncoding.java

```diff
@@ -137,18 +129,23 @@ public static byte[] compress(String value) {
     }
 
     public static byte[] compress(String value, float bufferRatio) {
-        BytesBuffer buf = LZ4Util.compress(encode(value), BLOCK_SIZE,
-                                           bufferRatio);
-        return buf.bytes();
+        try (BytesBuffer buf = LZ4Util.compress(encode(value), BLOCK_SIZE, bufferRatio)) {
+            return buf.bytes();
+        } catch (IOException e) {
```

Comment:
seems don't need to catch IOException

Reply:
the IOException throws by `LZ4Util.compress/decompress` & we just auto release the `BytesBuffer` when it used up, so if we don't catch exception, we need to throws out? (so as the `decompress`

Reply:
> yes just remove the catch, since the LZ4Util.compress already transfer IOException to BackendException

so shall we close the `ByteBuffer` when it used up? if we remove catch with `try(xx)`, we need throws the exception?

It means we just ignore the `ByteBuffer` close?

Reply:
yes the BytesBuffer don't need to close, it's just a memory buffer

Reply:
> yes the BytesBuffer don't need to close, it's just a memory buffer

OK, my previous consideration is that if the **buffer** is large(like compress a big file), we should recycle it in time to reduce the pressure of GC, but lack enough validation, so back off for now